app/main/db/queries.py

```diff
@@ -34,10 +39,30 @@ def browse_data(
     elif consignment_id:
         consignment = Consignment.query.get_or_404(consignment_id)
         validate_body_user_groups_or_404(consignment.consignment_bodies.Name)
-        browse_query = _build_consignment_filter_query(consignment_id)
+
+        if date_range is not None and len(date_range) > 0:
+            if date_filter_field is not None:
+                if date_filter_field.lower() == "date_last_modified":
+                    browse_query = _build_consignment_filter_query(
+                        consignment_id,
+                        date_range,
+                        date_filter_field="date_last_modified",
+                    )
+        else:
+            browse_query = _build_consignment_filter_query(consignment_id)
     else:
         browse_query = _build_browse_everything_query()
 
+    if not consignment_id and date_range is not None and len(date_range) > 0:
```

Comment:
```suggestion
    if not consignment_id and date_range:
```

Reply:
done the changes as requested in latest commit